ui/src/components/InputDialog/RoleDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import { useLedger } from '@daml/react';
+
+import { InputDialog } from './InputDialog';
+
+import { Template } from '@daml/types';
+import { RoleRequestTemplates, RoleKind } from '../../context/RolesContext';
+
+interface RequestProps<T extends RoleRequestTemplates> {
+  fields: any;
+  open?: boolean;
+  params: T;
+  request: Template<T, undefined, string>;
+  role: RoleKind;
+  onChange: (state: any) => void;
```

Comment:
If the `K` type parameter above works, then

```suggestion
  onChange: (state: K | null) => void;
```